### DISCUSSION TITLE
different regional HttpProxyHandlerFactory, lead to a big difference in cost time

### DISCUSSION THREAD

**Discussion by luckytom121** [relayrides/pushy#968] [Q&A]
hi, When I build the ApnsClient instance, I use an HttpProxyHandlerFactory proxy. I have prepared a Singapore proxy and a Chinese proxy. I want to compare the cost time from client.SendNotification to receiving PushNotificationResponse. However, it is found that the time of using The proxy in China is more than three times that of using the proxy in Singapore. I just used proxy from different regions, why the huge difference in time？

```java
final ApnsClient apnsClient = new ApnsClientBuilder()
    .setApnsServer(ApnsClientBuilder.DEVELOPMENT_APNS_HOST)
    .setSigningKey(ApnsSigningKey.loadFromPkcs8File(new File("/path/to/key.p8"),
            "TEAMID1234", "KEYID67890"))
    .setProxyHandlerFactory(HttpProxyHandlerFactory.fromSystemProxies(
            ApnsClientBuilder.DEVELOPMENT_APNS_HOST))
    .build();   

PushNotificationFuture sendFuture = client.sendNotification(message);
```

**Comment by jchambers** [comment]
> The proxy in China is more than three times that of using the proxy in Singapore… why the huge difference in time?

There are no APNs servers in China, and routes from China to the rest of the world can be slow.

**Reply by luckytom121** [reply]
Thank you very much for your reply. For China, is there any way to optimize the time consumption, or make it as small as possible？
